src/modules/CRUD/services/CRUDProvider.js

```diff
@@ -29,105 +33,185 @@ export class CRUDProvider extends React.Component {
   setMap = map => !this.isUnmount && this.setState({ map });
 
   getMapConfig = async () => {
+    const result = {};
     try {
-      const { results: mapConfig = {} } = await fetchMapConfig();
-      this.setState({ mapConfig });
+      const mapConfig = await fetchMapConfig();
+      result.mapConfig = mapConfig;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, code: e.message },
-      }));
+      result.error = e;
     }
+
+    const { mapConfig = {}, error } = result;
+    this.setState(({ errors }) => ({
+      mapConfig,
+      errors: {
+        ...errors,
+        mapConfig: error,
+      },
+    }));
+
+    return mapConfig;
   };
 
   getSettings = async () => {
+    const result = {};
     try {
       const settings = await fetchSettings();
-      this.setState({ settings });
+      result.settings = settings;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, code: e.message },
-      }));
+      result.error = e;
+    }
+
+    const { settings = {}, error } = result;
+
+    this.setState(({ errors }) => ({
+      settings,
+      errors: {
+        ...errors,
+        settings: error,
+      },
+    }));
+
+    return settings;
+  }
+
+  getFormattedError = props => {
+    const {
+      store,
+      error,
+      ids,
+    } = props;
+    const { errors: { [store]: errorStore } } = this.state;
+
+    const propsFromIds = Object.keys(ids);
+
+    // Remove error if it was before in the store
+    if (!(error instanceof Error)) {
+      return errorStore.filter(item => (
+        propsFromIds.every(prop => item[prop] !== ids[prop])
+      ));
     }
+
+
+    // Modify error if it was before in the store
+    const isErrorItemAlreadyExisting = errorStore.some(item => (
+      propsFromIds.every(prop => item[prop] === ids[prop])
+    ));
+
+    if (isErrorItemAlreadyExisting) {
+      return errorStore.map(item => (
+        propsFromIds.every(prop => item[prop] === ids[prop])
+          ? { error, ...ids }
+          : item
+      ));
+    }
+
+    // Else add error in the store
+    return [...errorStore, { error, ...ids }];
   }
 
   getFeaturesList = async layerId => {
+    const result = {};
     try {
-      const { results: featuresList = [] } = await fetchFeaturesList(layerId);
-      this.setState({ featuresList });
+      const { results: featuresList } = await fetchFeaturesList(layerId);
+      result.featuresList = featuresList;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, [state.featuresList.length]: true },
-      }));
+      result.error = e;
     }
-  }
+    const { featuresList = [], error = {} } = result;
+    const { errors } = this.state;
+
+    this.setState({
```

Comment:
`setState` is async ! you should use `errors` from `setState` callback argument.